guides/doc-Upgrading_and_Updating/topics/upgrading_a_connected_satellite_server.adoc

```diff
@@ -137,7 +137,7 @@ endif::[]
 +
 [options="nowrap"]
 ----
-# rpm -qa --last | grep kernel
+# rpm -q --last kernel
 ----
 . Optional: If a kernel update occurred since the last reboot, reboot the system:
```

Comment:
I wonder if we should suggest running `uname --kernel-release` to determine the running version.

Reply:
Good idea!
we could use two commads like this:
```
rpm -q --last kernel
uname --kernel-release
```
These are easy to type but the user has to do that extra step of finding the running release among the showed updates and confirm that it is the last update.
Or we could merge those two commands into one slighty more difficult to type but it shows exactly the information the user wants, and that is whether the newest kernel is currently running:
```
rpm -q --last kernel | head -n1 | grep -v "$(uname --kernel-release)"
```
This command outputs the newest installed kernel in a case that the kernel is not running and the system needs a reboot, or does not output anything in a case that the newest installed kernel is running.

Reply:
I'll defer to other writers, but I think we try to keep examples simple.

Reply:
So it should look something like this?

> 10. Check if the newest installed kernel is running:
> ```
> # rpm --query --last kernel | head -n 1
> # uname --kernel-release
> ```
> 11. Optional: If the kernel versions outputted from the two previous commands differ, reboot the system:
> ```
> # reboot
> ```

Reply:
I think the guidance is that there's at most one command per example. @maximiliankolb?

Reply:
For better readability, we generally place each argument in it's own line. Something like `rpm -q --last kernel | head -n 1 | grep -v "$(uname --kernel-release)"` is perfectly fine for me because you cannot remove options/lines or make alternations.

However, this needs to communicate that we expect exit code 1 but no stdout (unless my test system is broken).

Reply:
This is true.
Do you suggest the updated section could look like this?

> 10. Check if the newest installed kernel is running:
> ```
> # rpm --query --last kernel | head -n 1 | grep -v "$(uname --kernel-release)"
> ```
> If the command produces no output, it means that the newest installed kernel is currently running. Note: In such case, 
> the command exits with exit code 1.
> 11. Optional: If the last command produced output, reboot the system:
> ```
> # reboot
> ```

Reply:
Yes, something like that. You could also consider appending `|| true`: https://stackoverflow.com/a/61321426.

Reply:
@maximiliankolb Thank you, that seems like a valid solution. I got one more idea.
```
# rpm --query --last kernel | head -n 1 | grep $(uname --kernel-release)
```
this command produces output and exit code 0 if the newest installed kernel is currently running, or does not produce output and exits with code 1 if the system needs a reboot to switch to the newest kernel. Please let me know which alternative would be preferable to you. Either:
```
# rpm --query --last kernel | head -n 1 | grep $(uname --kernel-release)
```
Or:
```
# rpm --query --last kernel | head -n 1 | grep -v $(uname --kernel-release) || true
```

Reply:
I think the first solution is more elegant and easier to understand. Either way, we should let users know what to expect if everything is working as intended.

Reply:
I am quite hesitant to essentially provide shell scripts in documentation.

What comes to mind is that I think katello-host-tools has a mechanism to determine whether a reboot is needed. Perhaps that would be a thing to look into. Looks like it's here: https://github.com/FrostyX/tracer/blob/4b8d869fb4556efaa4c618c3c7102f25239abc95/tracer/resources/tracer.py#L133-L144 (which is essentially the same logic).

For reference, on Debian it's: if `/var/run/reboot-required` exists, you need to reboot. Suse appears to have `/var/run/reboot-needed`.

@upadhyeammit perhaps this should be included in foreman-maintain instead?